### ISSUE TITLE
hover comment, there is a chance that the comment of the first function will not be displayed

### ISSUE THREAD

**Issue by q962** [microsoft/vscode-cpptools#9137]
Bug type: Language Service

**Describe the bug**
- OS and Version:  windows_NT x64 10.0.19044
- VS Code Version: 1.66.0
- C/C++ Extension Version: v1.9.7
- Hover comments for the first function sometimes don't show up

**Steps to reproduce**
1. Move the mouse over the function name

**Expected behavior**
The first function displays the comment normally

**Code sample and logs**

no error
```
cpptools/textEditorSelectionChange
cpptools/textEditorSelectionChange
cpptools/activeDocumentChange: <paht>/g_file.expand.h
textDocument/hover: <paht>/g_file.expand.h (id: 18)
textDocument/definition: <paht>/g_file.expand.h (id: 19)
textDocument/hover: <paht>/g_file.expand.h (id: 20)
textDocument/hover: <paht>/g_file.expand.h (id: 21)
textDocument/hover: <paht>/g_file.expand.h (id: 22)
textDocument/hover: <paht>/g_file.expand.h (id: 23)
```

**Additional context**

https://user-images.githubusercontent.com/15686075/161551341-3a987a33-4172-46be-87ba-a527aa91270d.mp4

**Comment by michelleangela** [comment]
@q962 
I'm not able to reproduce the case where the first function of the comment does not get displayed on hover. This could happen if there is a delay in IntelliSense updating/parsing, but once IntelliSense is done updating/parsing a source file, it should send the correct info for hover.

Does this issue also happen for simple projects or only for certain projects? Is it always only on the first function? Are you using the extension on a remote machine?

**Comment by q962** [comment]
[vscode_Cpp_test.zip](https://github.com/microsoft/vscode-cpptools/files/8414221/vscode_Cpp_test.zip)

100% trigger

**Comment by michelleangela** [comment]
Thank you for the sample code. This is a duplicate of https://github.com/microsoft/vscode-cpptools/issues/8320.

Using the example code in screenshot below, if a comment is placed above the macro, the following function will get the comment instead of the macro getting the comment, which is the same behavior as issue https://github.com/microsoft/vscode-cpptools/issues/8320.
![image](https://user-images.githubusercontent.com/38734287/161831761-78bc1efe-45b5-450e-a4e6-22aac1426298.png)

Closing issue as duplicate.
